sql/item_sum.cc

```diff
@@ -2197,6 +2216,340 @@ double Stddev::result(bool is_sample_variance)
 }
 
 
+void Regression_result::recurrence_next(double nr_y, double nr_x)
+{
+
+  if (N++)
+  {
+    sx+= nr_x;
+    sy+= nr_y;
+    double diff = nr_x - sxx.get_m();
+    sxx.recurrence_next(nr_x);
+    syy.recurrence_next(nr_y);
+    sxy= sxy + diff * (nr_y - syy.get_m());
+  }
+  else
+  {
+    sx= nr_x;
+    sy= nr_y;
+    sxx.recurrence_next(nr_x);
+    syy.recurrence_next(nr_y);
+  }
+}
+
+void Regression_result::remove(double nr_y, double nr_x)
+{
+  if (N > 0)
+  {
+    sx-= nr_x;
+    sy-= nr_y;
+    sxx.remove(nr_x);
+    double diff = nr_x - sxx.get_m();
+    sxy= sxy - diff * (nr_y - syy.get_m());
+    syy.remove(nr_y);
+    N--;
+  }
```

Comment:
Same as a previous comment. Let's use a DBUG_ASSERT to ensure we never get called with N == 0.